ewatercycle/forcing/_lisflood.py

```diff
@@ -0,0 +1,110 @@
+"""Forcing related functionality for lisflood"""
+
+from pathlib import Path
+from typing import Optional
+
+from esmvalcore.experimental import get_recipe
+
+from ..util import data_files_from_recipe_output, get_extents, get_time
+from ._default import DefaultForcing
+from .datasets import DATASETS
+
+
+class LisfloodForcing(DefaultForcing):
+    """Container for lisflood forcing data."""
+
+    # TODO check whether start/end time are same as in the files
+    def __init__(
+        self,
+        start_time: str,
+        end_time: str,
+        directory: str,
+        shape: str,
+        PrefixPrecipitation: Optional[str] = 'pr.nc',
+        PrefixTavg: Optional[str] = 'tas.nc',
+        PrefixE0: Optional[str] = 'e0.nc',
+        PrefixES0: Optional[str] = 'es0.nc',
+        PrefixET0: Optional[str] = 'et0.nc',
+    ):
+        """
+            PrefixPrecipitation: Path to a NetCDF or pcraster file with precipitation data
+            PrefixTavg: Path to a NetCDF or pcraster file with average temperature data
+            PrefixE0: Path to a NetCDF or pcraster file with potential evaporation rate from open water surface data
+            PrefixES0: Path to a NetCDF or pcraster file with potential evaporation rate from bare soil surface data
+            PrefixET0: Path to a NetCDF or pcraster file with potential (reference) evapotranspiration rate data
+        """
+        super().__init(start_time, end_time, directory, shape)
+        self.PrefixPrecipitation = PrefixPrecipitation
+        self.PrefixTavg = PrefixTavg
+        self.PrefixE0 = PrefixE0
+        self.PrefixES0 = PrefixES0
+        self.PrefixET0 = PrefixET0
+
+    @classmethod
+    def generate(  # type: ignore
+        cls,
+        dataset: str,
+        start_time: str,
+        end_time: str,
+        shape: str,
+        extract_region: dict = None,
+    ) -> 'LisfloodForcing':
+        """
```

Comment:
Maybe you could do something like:
```suggestion
        super().generate.__doc__ + """
```
to generate the complete documentation here?